beater/server.go

```diff
@@ -101,13 +101,13 @@ func createHandler(p processor.Processor, config Config, publish successCallback
 
 		// Limit size of request to prevent for example zip bombs
 		limitedReader := io.LimitReader(reader, config.MaxUnzippedSize)
-		err = p.Validate(limitedReader)
+		payload, err := p.Validate(limitedReader)
 		if err != nil {
 			sendError(w, r, 400, fmt.Sprintf("Data Validation error: %s", err), true)
 			return
 		}
 
-		list := p.Transform()
+		list := payload.Transform()
```

Comment:
I would not expose the `payload` here, but rather handle it as an interna of `processor`. The publicly known and agreed interface is `Processor` which offers a `Validate(..)` and a `Transform(..)` method.